### DISCUSSION TITLE
Is it possible to add translation function? Not online but just as open webui function

### DISCUSSION THREAD

**Discussion by Llamadouble999q** [open-webui/open-webui#4081] [General]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by atgehrhardt** [comment]
I am a bit confused at what you are asking here. 

Are you asking if there is a way to implement an offline translation library? If so, you can do this with a tool, a function, or a pipeline, however there aren’t many great offline translation libraries to my knowledge.

Please fill out the template when you submit issues.

**Reply by Llamadouble999q** [reply]
Does it translate my chat right on the page of open webui? Like chatgpt  translate plugin it shows translation under the model’s answer

**Comment by justinh-rahb** [comment]
https://openwebui.com/f/iamg30/libretranslate_filter

**Reply by Llamadouble999q** [reply]
But Libre translate doesnt work for me it sends me error message only do you know how to set libretrnalsate via docker and add other languages?

**Comment by silentoplayz** [comment]
> https://openwebui.com/f/iamg30/libretranslate_filter

This function requires further setup with [LibreTranslate](https://github.com/LibreTranslate/LibreTranslate/tree/main) integration into Open WebUI to utilize. And while it's not complicated to setup, there still isn't any documentation for users of Open WebUI to set this up themselves.

https://openwebui.com/f/justinrahb/google_translate/ on the other hand, should work right out of the box, yeah? 😍

**Reply by Llamadouble999q** [reply]
But Libre translate doesnt work for me it sends me error message only do you know how to set libretrnalsate via docker and add other languages?
